### PR TITLE
Create a mock kafka factory to enable unit tests without Kafka

### PR DESCRIPTION
**Pull request by Pchelolo**
cc @wikimedia/services

### CODE REVIEW

File: lib/kafka_factory.js [hunk 1]
```diff
@@ -291,4 +291,17 @@ class KafkaFactory {
         return this._createProducerOfClass(GuaranteedProducer, logger);
     }
 }
-module.exports = KafkaFactory;
+
+let GLOBAL_FACTORY;
```
Comment (d00rman) [inline lib/kafka_factory.js]: Would it not be better to have the factory to use as a param to `create` instead of global vars?

File: .travis.yml [hunk 2]
```diff
@@ -21,8 +21,7 @@ addons:
 
 before_script:
 - npm run install-kafka
-- npm run start-kafka
 
-script: npm run coverage && (npm run coveralls || exit 0)
+script: npm test && npm run start-kafka && npm run coverage && (npm run coveralls || exit 0)
```
Comment (d00rman) [inline .travis.yml]: `npm run coverage` will run `npm test` again, won't it?

Comment (Pchelolo) [reply .travis.yml]: coverage runs 'mocha' with no MOCK_KAFKA env variable - thus it's an integration test.

File: lib/kafka_factory.js (hunk 1 above)
Comment (Pchelolo) [reply lib/kafka_factory.js]: Param to create? We need to store the global factory somewhere if we want to replace it from tests preparation code

**Pull request merged**